enhancements/installer/signal-cluster-deletion.md

```diff
@@ -0,0 +1,121 @@
+---
+title: signal-cluster-deletion
+authors:
+  - "@abutcher"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-03-30
+last-updated: 2020-03-30
+status: provisional
+---
+
+# Signal Cluster Deletion
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Operators can create external resources in clouds and other services that have no opportunity to be cleaned up when a cluster is uninstalled, causing extra work for cluster maintainers to track down and remove these resources. Signalling cluster deletion would allow operators to clean up external resources during uninstallation and would ensure that total cleanup is an automatic process.
+
+## Motivation
+
+### Goals
+
+Signal that cluster deletion has begun through deletion of a new CRD `ClusterAlive` and wait for that object to be deleted during uninstall before continuing with removing cluster resources such as instances, storage and networking.
+
+### Non-Goals
+
+Attaching finalizers to operator resources based on the `ClusterAlive` object which would facilitate removal of operator resources.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints
+
+#### API
+
+Introduce a new CRD `ClusterAlive` from which operator resources could attach a finalizer initiating cascading delete when the `ClusterAlive` resource is deleted.
+
+```yaml
+apiVersion: v1
+kind: ClusterAlive
+metadata:
+  name: cluster
+  namespace: openshift-config
```

Comment:
Does this need to be namespaced or should it be global?

Reply:
I don't think this needs to be namespaced.

Reply:
Namespace removed in last update.

Reply:
I don't love non-namespaced things. Especially if we are going to make it a generic Kind like `Alive` instead of `ClusterAlive` then having something namespaced I could use the same way for me own purpose would be nice.

Reply:
Re-added namespace to the object.